adaptive_navigation/lib/scaffold.dart

```diff
@@ -52,105 +46,132 @@ class AdaptiveNavigationScaffold extends StatefulWidget {
+                    (d) => BottomNavigationBarItem(
+                  icon: Icon(d.icon),
+                  title: Text(d.title),
                 ),
-                for (var d in widget.destinations)
-                  ListTile(
-                    leading: Icon(d.icon),
-                    title: Text(d.title),
-                    selected:
-                        widget.destinations.indexOf(d) == widget.currentIndex,
-                    onTap: () => _destinationTapped(d),
-                  ),
-              ],
-            ),
+              ),
+            ],
+            currentIndex: widget.currentIndex,
+            onTap: widget.onNavigationIndexChange,
           ),
-          VerticalDivider(
-            width: 1,
-            thickness: 1,
-            color: Colors.grey[300],
+          floatingActionButton: widget.floatingActionButton,
+        );
+      case NavigationType.navigationRail:
+        // Show a Scaffold with a body containing a NavigationRail.
+        return Scaffold(
+          appBar: AppBar(
+            title: widget.title,
           ),
-          Expanded(
-            child: Scaffold(
-              appBar: AppBar(),
-              body: widget.body,
-              floatingActionButton: widget.floatingActionButton,
-            ),
+          body: Row(
+            children: [
+              NavigationRail(
+                leading: widget.floatingActionButton,
+                destinations: [
+                  ...widget.destinations.map(
+                        (d) => NavigationRailDestination(
+                      icon: Icon(d.icon),
+                      label: Text(d.title),
+                    ),
+                  ),
+                ],
+                selectedIndex: widget.currentIndex,
+                onDestinationSelected: widget.onNavigationIndexChange ?? (_) {},
+              ),
+              VerticalDivider(
+                width: 1,
+                thickness: 1,
+                color: Colors.grey[300],
+              ),
+              Expanded(
+                child: widget.body,
+              ),
+            ],
           ),
-        ],
-      );
-    }
-
-    // Show a navigation rail
-    if (_isMediumScreen(context)) {
-      return Scaffold(
-        appBar: AppBar(
-          title: widget.title,
-        ),
-        body: Row(
+        );
+      case NavigationType.drawer:
+        // Show a Row containing a Drawer and Scaffold.
+        return Row(
           children: [
-            NavigationRail(
-              leading: widget.floatingActionButton,
-              destinations: [
-                ...widget.destinations.map(
-                  (d) => NavigationRailDestination(
-                    icon: Icon(d.icon),
-                    label: Text(d.title),
+            Drawer(
+              child: Column(
+                children: [
+                  DrawerHeader(
+                    child: Center(
+                      child: widget.title,
```

Comment:
Hmm Why is the title put into the drawer instead of AppBar?

Reply:
Not sure, I will bring this up in the design review